### DISCUSSION TITLE
Is there any reason to repeat a dataset multiple times over an epoch?

### DISCUSSION THREAD

**Discussion by abetteridea** [kohya-ss/sd-scripts#678] [Q&A]
I wonder if there is any reason to repeat a dataset of 20-30 images multiple times within an epoch?

As @kohya-ss said [here](https://github.com/kohya-ss/sd-scripts/discussions/182#discussioncomment-4946693):

> There are two purpose for repeats.
> 
> 1. Balance the number of training images to the regularization images. The number of regularization images is larger than the training, so it is required to repeat training images for using all regularization images in the epoch.
> 2. Control 'weight' over folders. If you have high quality images and low quality images, you can set higher number of repeats for high quality images, and lower for low quality.

And yet, many people are repeating their datasets 10-40 times within an epoch, even when they only have one img folder and no reg images. Is there any benefit in doing so?

**Comment by TingTingin** [comment]
repeats and epochs mean the same thing if you have only one folder

**Comment by abetteridea** [comment]
> repeats and epochs mean the same thing if you have only one folder

They don't for _Cosine with restarts_ with default settings, for instance.

And as far as I understand the end of an epoch is a different process (in terms of code) compared to the end of a repeat.
